### ISSUE TITLE
StepVotesMsg.Copy fails with panic after UnmarshalStepVotes is called but err is not checked

### ISSUE THREAD

**Issue by devtomnl** [dusk-network/dusk-blockchain#612]
```
time="2020-08-04T23:15:55+02:00" level=info msg="initializing node..." app=dusk prefix=main
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xa16d36]

goroutine 98309 [running]:
github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/message.StepVotesMsg.Copy(0xc0002802d0, 0x81, 0x81, 0x533, 0xa, 0x1753dc0, 0x20, 0x20, 0x0, 0x0, ...)
	/opt/gocode/src/github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/message/agreement.go:122 +0x2a6
github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/message.Clone(0x1088ea0, 0xc001161300, 0xc000db8b01, 0xc00168400a)
	/opt/gocode/src/github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/message/message.go:79 +0xe0
github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/eventbus.(*CallbackListener).Notify(0xc00019c460, 0x1088ea0, 0xc001161300, 0x1, 0xc000d46001)
	/opt/gocode/src/github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/eventbus/listener.go:35 +0x44
github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/eventbus.(*multiListener).Forward(0xc000046bc0, 0xd74920, 0x1088ea0, 0xc001161300)
	/opt/gocode/src/github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/eventbus/listener.go:180 +0x13c
created by github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/eventbus.(*EventBus).Publish
	/opt/gocode/src/github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/eventbus/publisher.go:25 +0x81
```

**Comment by devtomnl** [comment]
The copy method follows the Safe interface that does not account for errors.
Since the method could fail, shall the interface be updated to return err ?

```
// Safe is used for protection from race conditions
type Safe interface {
	// Copy performs a deep copy of an object
	Copy() (Safe, error)
}
```

**Comment by devtomnl** [comment]
That wont be possible without changing things like this method:
```
// UnmarshalAgreementMessage unmarshal a network inbound Agreement
func UnmarshalAgreementMessage(r *bytes.Buffer, m SerializableMessage) error {
	aggro := newAgreement()
	(...)
	m.SetPayload(*aggro)
	return nil
}
```

**Comment by autholykos** [comment]
Closed via #608
